BUILD_ASE/TEST/diels/grad.py

```diff
@@ -14,31 +15,54 @@
 argv1 = sys.argv[1]
 argv2 = sys.argv[2]
 #print 'argv1: ',argv1
-fname = 'scratch/structure'+argv1
-slab = read(fname)
-
-#slab.set_calculator(EMT())
-slab.set_calculator(NWChem())
-slab.calc.label = 'scratch/nwchem'+argv1
-inputfile = 'nwchem'+argv1+'.nw'
-outputfile = 'nwchem'+argv1+'.out'
-##slab.calc.command = 'cd scratch; mpirun -np '+argv2+' nwchem '+inputfile+' > '+outputfile
-slab.calc.command = 'cd scratch; mpirun --mca btl ^openib -np '+argv2+' nwchem '+inputfile+' > '+outputfile
-print 'NWChem command: ',slab.calc.command
-os.system('rm -f '+slab.calc.label+'.scrdir')
-os.system('ln -s $PBSTMPDIR '+slab.calc.label+'.scrdir')
-
-slab.calc.parameters.xc = 'B3LYP'
-#slab.calc.parameters.basis = 'LANL2DZ ECP'
-#slab.calc.parameters.ecp = 'LANL2DZ ECP'
-slab.calc.reset()
-
-energy = - slab.get_potential_energy()
-grads = - slab.get_forces()
+os.system("cp scratch/structure"+argv1+" scratch/structure"+argv1+".xyz")
+fname = 'scratch/structure'+argv1+'.xyz'
+atoms = read(fname)
+folder = 'scratch/qchem'+argv1
+
+#atoms.set_calculator(EMT())
+calc = QChem(
+    jobtype='FORCE',
+    method='B3LYP',
+    basis='6-31G*',
+    nt=8,
+)
+
+atoms.calc = calc
+
+atoms.calc.label = 'scratch/qchem'+argv1
+inputfile = 'qchem'+argv1+'.inp'
+outputfile = 'qchem'+argv1+'.out'
+##atoms.calc.command = 'cd scratch; mpirun -np '+argv2+' nwchem '+inputfile+' > '+outputfile
+# atoms.calc.command = 'cd scratch; mpirun --mca btl ^openib -np '+argv2+' nwchem '+inputfile+' > '+outputfile
+atoms.calc.command = f'cd scratch; qchem -nt 8 '+inputfile+' > '+outputfile
+print('QChem command: ',atoms.calc.command)
+# os.system('ln -s $PBSTMPDIR '+atoms.calc.label+'.scrdir')
+
+# atoms.calc.parameters.xc = 'B3LYP'
+#atoms.calc.parameters.basis = 'LANL2DZ ECP'
+#atoms.calc.parameters.ecp = 'LANL2DZ ECP'
+atoms.calc.reset()
+
+cwd = os.getcwd()
+
+if not os.path.exists(folder):
+    os.mkdir(folder)
+os.chdir(folder)
+
+energy = - atoms.get_potential_energy()
+grads = - atoms.get_forces()
 #print grads
 f = open('scratch/GRAD'+argv1, 'w')
 f.write(str(energy))
 f.write('\n')
 f.write(str(grads))
 f.write('\n')
 f.close()
+
+shutil.copy2('scratch/GRAD'+argv1, cwd+'/scratch')
+os.chdir(cwd)
+
+os.system('rm -rf '+folder)
```

Comment:
This is asking for trouble

Reply:
@charnley Are you primarily concerned about `rm -rf` in a system call in general or something specific here? How would you recommend removing the temporary scratch directories?